espnet2/tts/prodiff/prodiff.py

```diff
@@ -661,9 +672,13 @@ def _forward(
             h_masks = None
 
         if self.decoder_type == "diffusion":
-            before_outs = self.decoder(
-                hs, ys, h_masks, is_inference
-            )  # (B, T_feats, odim)
+            if not is_inference:
```

Comment:
Is this change really necessary?
The code at line https://github.com/espnet/espnet/blob/513318e31e46b0a98eabe0bab8948d3cec71db25/espnet2/tts/prodiff/denoiser.py#L237
performs only inference, so it seems unnecessary to me. 
Could you please check and see if there is any difference between your proposed code and the original? If not, please revert it, to approve your PR.

Reply:
Oh, my bad. In my trials to get the teacher forcing working ASAP, I just removed the "is_inference" parameter being passed because teacher forcing was used using training. Later when I was trying to write a clean solution, somehow  I skipped this line.

I'll fix this and raise another pull request.